src/plugin.ts

```diff
@@ -20,8 +20,12 @@ async function main() {
   const ctx: Context = { typeMap, options, utils };
 
   const files = await Promise.all(
-    request.protoFile.map(async (file) => {
-      const [path, code] = generateFile(ctx, file);
+    request.fileToGenerate.map(async (fileName) => {
+      const descriptor = request.protoFile.find(desc => desc.name == fileName);
+      if (descriptor === undefined) {
+        throw new Error(`Cannot find descriptor for file-to-generate: ${fileName}`);
+      }
+      const [path, code] = generateFile(ctx, descriptor);
```

Comment:
Cool, this looks good. Just to make sure `src/plugin.ts` and `integration/codegen.ts` stay in sync (I've had bugs come up from them drifting before), how about:

* Add a new `function protoFilesToGenerate(request): FileDescriptorProto[]` in `src/plugin.ts` that returns `request.protoFile.filter(f => request.filesToGenerate.includes(f.name));`
* Make line 23 in `src/plugin.ts` be `protoFilesToGenerate(request).map(async file =>`
* Make line 34 in `codegen.ts` be `for (let file of protoFilesToGenerate(request)) {`

?

Reply:
That sounds reasonable, will make the change.